pkg/stub/watch.go

```diff
@@ -0,0 +1,48 @@
+package stub
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func Watch(getWatchInterface func() (watch.Interface, error), handleChange func()) {
+	watchInterface, err := getWatchInterface()
+	if err != nil {
+		return
+	}
+	events := watchInterface.ResultChan()
+	defer func() {
+		if watchInterface != nil {
+			watchInterface.Stop()
+		}
+	}()
+	willHandle := false
+	for {
+		select {
+		case _, ok := <-events:
+			if !ok {
+				watchInterface.Stop()
+				watchInterface, err = getWatchInterface()
+				if err != nil {
+					return
+				}
+				events = watchInterface.ResultChan()
+				continue
+			}
+			if willHandle {
+				continue
+			}
+
+			// this makes sure that there will be only one call of handleChange in case there are more
+			// events notifying about change in the same time
```

Comment:
right, now I understand.

In this case, would it be better if we keep the timestamp of when the last event was received, and if the time of the new event is too close to the previous one, we just ignore it? It's the same idea, but might be a bit easier to understand.

Reply:
IMHO we would still need the goroutine to execute the handle function after the 1 second. if we execute it with the first event, the later changes (which will happen in that 1 second timeframe) might be ignored (not reflected in mobile-services.json files).